int32 job_num
Operation[] operations